go/libraries/doltcore/env/actions/reset.go

```diff
@@ -136,26 +136,31 @@ func ResetHardTables(ctx context.Context, dbData env.DbData, cSpecStr string, ro
 	return resetHardTables(ctx, dbData, cSpecStr, roots)
 }
 
-func ResetHard(ctx context.Context, dEnv *env.DoltEnv, cSpecStr string, roots doltdb.Roots) error {
+// ResetHard resets the working, staged, and head based on the given parameters.
+// The reset can be performed on a non-current branch and working set.
+// Returns an error if the reset fails.
+func ResetHard(
+	ctx context.Context,
+	dEnv *env.DoltEnv,
+	cSpecStr string,
+	roots doltdb.Roots,
+	headRef *ref.DoltRef,
```

Comment:
Why is this a pointer?

Reply:
Good question, not sure. No longer a pointer.

In general, which objects should/shouldn't be passed as pointers?
You mentioned `doltdb.Roots` as an example of a struct that we always pass by value. How are we deciding this?

Reply:
I'll come talk about it